spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/source/SpringConfigurationPropertySources.java

```diff
@@ -91,10 +91,10 @@ public boolean hasNext() {
 		@Override
 		public ConfigurationPropertySource next() {
 			ConfigurationPropertySource next = fetchNext();
-			this.next = null;
 			if (next == null) {
```

Comment:
Oops. Looks like we have a test missing as well there.

Reply:
@snicoll This change is just for avoiding a `null` assignment to `this.next` when `next` is `null` as `this.next` is already `null`, so it should not change practically nothing. Am I missing something?

Reply:
You don't but I certainly did 😂